src/test/java/me/creighton/encodedid/TestBigIntegerEncoder.java

```diff
@@ -279,28 +284,44 @@ public void encodingTests () {
     sc = encoder5.encodeId(d);
     e = encoder4.decodeId(s);
     c = encoder5.decodeId(sc);
-    System.out.printf("%d encodes as %s; with check char as %s; decodes to %d\n", d, s, sc, e);
+    showMessage(String.format("%d encodes as %s; with check char as %s; decodes to %d", d, s, sc, e));
     assertEquals(d,e);
     assertEquals(d,c);
 
-
-    try {
-      d = bTrillion;
-      s = encoder2.encodeId(d);
-      e = encoder2.decodeId("DYB2-C9DM-QW");
-      System.out.printf("1 trillion with bad check character encodes as %s; decodes to %d\n", s, e);
-    } catch (EncodedIdException ex) {
-      System.out.printf("\n!!! Caught exception: %s", ex.getMessage());
-    }
-    try {
-      d = bTrillion;
-      s = encoder5.encodeId(d);
-      e = encoder5.decodeId("oxsrN185Mkz;W");
-      System.out.printf("1 trillion with bad check character encodes as %s; decodes to %d\n", s, e);
-    } catch (EncodedIdException ex) {
-      System.out.printf("\n!!! Caught exception: %s", ex.getMessage());
-    }
+    d = bTrillion;  // test two encoders with 10^^12; test bad check char handling
+    final String tmp1, tmp2;
+    EncodedIdException thrown;
+
+    showMessage("\n");
+    s = encoder2.encodeId(d);
+    showMessage(String.format("Encoded 1 trillion with check char: %s", s));
+    tmp1 = "DYB2C9DMQX";  // Correct encoding is DYB2C9DMQG
+    thrown =
+            assertThrows(EncodedIdException.class, () -> {
+              encoder2.decodeId(tmp1);  // Should be: DYB2C9DMQG
+            }, "EncodedIdException error (invalid check digit) expected!!!");
+    showMessage(thrown.getMessage());
+    e = encoder2.decodeId(s);
+    assertEquals(d, e);
+    showMessage(String.format("The encoded ID, including check digit: %s We gave it %s", s, tmp1));
+
+    showMessage("\n");
+    s = encoder5.encodeId(d);
+    showMessage(String.format("Encoded 1 trillion with check char: %s", s));
+    tmp2 = "gglq&Di.";  // Correct encoding is gglq&Di;
+    thrown =
+            assertThrows(EncodedIdException.class, () -> {
+              encoder2.decodeId(tmp2);  // Should be: gglq&Di;
+            }, "EncodedIdException error (invalid check digit) expected!!!");
+    showMessage(thrown.getMessage());
+    e = encoder5.decodeId(s);
+    assertEquals(d, e);
+    showMessage(String.format("The encoded ID, including check digit: %s We gave it %s", s, tmp2));
 
   }
 
+  private void showMessage (String msg) {
+    if (SHOW)
+      System.out.printf("%s\n", msg);
```

Comment:
Minor feedback - code style recommends explicit curly braces to avoid programmer mistakes with mishandling implicit scopes:

https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces


```suggestion
    if (SHOW) {
      System.out.printf("%s\n", msg);
    }
```